crabs/detection_tracking/models.py

```diff
@@ -67,17 +77,89 @@ def on_train_epoch_end(self):
         self.logger.log_metrics(
             {"train_loss": avg_loss}, step=self.current_epoch
         )
-
-        # Reset the training_step_outputs for the next epoch
         self.training_step_outputs = {
             "total_training_loss": 0.0,
             "num_batches": 0,
         }
 
+    def on_validation_epoch_end(self):
```

Comment:
This is tricky. The reason why I ended up need to use precision and not loss to train optuna is because the output during `model(input)` for training which gives loss not giving loss during eval. It returns the prediction output. It's possible but quite a pain. So maybe need to do next TI

Reply:
aah that is annoying! ok let's keep an eye on it, and consider attacking it next TI